src/Platform/Microsoft.Testing.Platform/Hosts/TestHostBuilder.cs

```diff
@@ -236,8 +237,30 @@ public async Task<ITestHost> BuildAsync(
             Logging.AddProvider((_, _) => loggingState.FileLoggerProvider);
         }
 
-        // Register the server mode log forwarder if needed. We follow the console --diagnostic behavior.
         ICommandLineOptions commandLineOptions = serviceProvider.GetCommandLineOptions();
+
+        // setting the timeout
+        if (commandLineOptions.IsOptionSet(PlatformCommandLineProvider.TimeoutOptionKey) && commandLineOptions.TryGetOptionArgumentList(PlatformCommandLineProvider.TimeoutOptionKey, out string[]? args))
+        {
+            string arg = args[0];
+            int size = arg.Length;
+            if (!float.TryParse(arg[..(size - 1)], out float value))
+            {
+                throw new InvalidOperationException(PlatformResources.PlatformCommandLineTimeoutArgumentErrorMessage);
```

Comment:
We can use `throw ApplicationStateGuard.Unreachable()`, because this is already validated so this would be some unexpected state.

```suggestion
                throw ApplicationStateGuard.Unreachable();
```